session/schema_amender.go

```diff
@@ -657,13 +654,21 @@ func (s *SchemaAmender) AmendTxn(ctx context.Context, startInfoSchema tikv.Schem
 	infoSchemaAtStart := startInfoSchema.(infoschema.InfoSchema)
 	infoSchemaAtCheck := change.LatestInfoSchema.(infoschema.InfoSchema)
 
+	var amendableType = []uint64{
+		uint64(model.ActionAddColumn),
+		uint64(model.ActionDropColumn),
+		uint64(model.ActionDropIndex),
+		uint64(model.ActionAddIndex),
+		uint64(model.ActionModifyColumn),
+	}
+
 	// Collect amend operations for each table by physical table ID.
 	var needAmendMem bool
 	amendCollector := newAmendCollector()
 	for i, tblID := range change.PhyTblIDS {
 		actionType := change.ActionTypes[i]
 		// Check amendable flags, return if not supported flags exist.
-		if actionType&(^amendableType) != 0 {
+		if !slices.Contains(amendableType, actionType) {
```

Comment:
I think it will influences performance when amend txn opened. But after MDL, the amend txn is no longer recommended. ref https://github.com/pingcap/tidb/issues/39196